tests/iterators/test_pdf_page_extractor.py

```diff
@@ -0,0 +1,62 @@
+import io
+import os
+from typing import List
+
+import pytest
+from PIL import Image
+
+import pixeltable as pxt
+from pixeltable.iterators import PdfPageExtractor
+
+
+def find_pdfs(path: str, limit: int = 50, recursive: bool = True) -> List[str]:
+    """Find PDF files in a directory up to a certain limit."""
+    pdf_paths = []
+    if recursive:
+        for root, _, files in os.walk(path):
+            for file in files:
+                if file.lower().endswith('.pdf'):
+                    pdf_paths.append(os.path.join(root, file))
+                    if len(pdf_paths) >= limit:
+                        return pdf_paths
+    else:
+        for file in os.listdir(path):
+            if file.lower().endswith('.pdf'):
+                pdf_paths.append(os.path.join(path, file))
+                if len(pdf_paths) >= limit:
+                    break
+    return pdf_paths
+
+
+class TestPdfPageExtractor:
+    @pytest.mark.parametrize("limit", [2])  # Only test a few PDFs to keep runtime low
+    def test_pdf_page_extraction(self, limit: int) -> None:
+        pdfs = find_pdfs("tests/data/documents", limit=limit)
```

Comment:
The `find_pdfs` function shouldn't be necessary; just use

`glob.glob('tests/data/documents/**/*.pdf')`

Better yet, use the existing `utils.get_documents()` which uses more robust path resolution, and filter on pdfs:

`pdfs = [path for path in get_documents() if path.endswith('.pdf')]`